File: monsoon3.0/src/main/java/wtf/monsoon/impl/event/EventPacket.java
package wtf.monsoon.impl.event;

import lombok.Data;
import lombok.NonNull;
import net.minecraft.network.Packet;
import wtf.monsoon.api.event.Event;

@Data
public class EventPacket extends Event {

    @NonNull
    public Packet packet;
    public boolean cancelled;
    @NonNull
    public Direction direction;


    public enum Direction {
        SEND, RECEIVE
    }

}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/event/EventPacket.kt
package wtf.monsoon.client.event

import me.bush.eventbuskotlin.Event
import net.minecraft.network.Packet
import spritz.api.annotations.Excluded

data class EventPacket(@JvmField var packet: Packet<*>, @Excluded val direction: PacketDirection) : Event() {

    @Excluded
    override val cancellable: Boolean = true

    @Excluded
    enum class PacketDirection {
        SEND, RECEIVE
    }
}